### PR TITLE
`Communication`: Hide notifications for active channel on lecture and exercise page

### PR DESCRIPTION
**Pull request by lennart-keller**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/client-tests/).

### Motivation and Context

Notifications are shown for the active channel on the lecture/exercise detail pages, even though the channel is open.

### Description

Notifications for the open channel are temporarily disabled as long as the sidebar is visible

### Steps for Testing

Prerequisites:
- 2 Students
- 1 Course with messaging enabled

1. Go to the same exercise/lecture details page with both students
2. Write messages as student 1
3. Student 2 should see the message, but no notification
4. Go to the messages tab with student 1 and write a message in a different channel (which student 2 is also a member of).
5. Student 2 should see a notification for this

#### Performance Review
- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [discussion-section.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5617/latest/artifact/shared/Coverage-Report-Client-Tests/app/overview/discussion-section/discussion-section.component.ts.html) | 86.77% | ✅ |
| [notification.service.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5617/latest/artifact/shared/Coverage-Report-Client-Tests/app/shared/notification/notification.service.ts.html) | 75% | ✅ |
| [notification-settings.service.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5617/latest/artifact/shared/Coverage-Report-Client-Tests/app/shared/user-settings/notification-settings/notification-settings.service.ts.html) | 94.44% | ✅ |

### Screenshots

No UI changes

### CODE REVIEW

**Review by Kroko-fant — DISMISSED** [review]
Tested in a testing session worked as expected

**Review by basak-akan — APPROVED** [review]
Tested in the testing session on TS2, notifications for the open channel disabled 👍

**Review by rstief — DISMISSED** [review]
Tested in session on ts2

**Review by MaximilianAnzinger — APPROVED** [review]
Code lgtm

File: src/main/webapp/app/overview/discussion-section/discussion-section.component.ts [hunk 1]
```diff
@@ -91,6 +93,7 @@ export class DiscussionSectionComponent extends CourseDiscussionDirective implem
      * on leaving the page, the modal should be closed
```
Comment (jakubriegel) [inline src/main/webapp/app/overview/discussion-section/discussion-section.component.ts:93]: This comment is not up to date now

**Review by jakubriegel — COMMENTED** [review]
Small note about one comment

**Review by krusche — CHANGES_REQUESTED** [review]
can you write some client tests to verify the new behavior

**Pull request closed without merging**